### PR TITLE
Split checkbox switch mode to new functional widget & update themes

### LINKED ISSUE #796: Take switch out of checkbox
This issue is concerned with removing the switch `mode` from checkbox.
We should create a separate switch widget and ensure that it is both theme-able and accessible.
It should be based on a checkbox input to ensure a11y compliance.

### PR DESCRIPTION
**Pull request by ericos**
**Type:**  feature

The following has been addressed in the PR:

* [x] There is a related issue
* [x] All code matches the [style guide](https://github.com/dojo/framework/blob/master/STYLE.md)
* [x] Unit or Functional tests are included in the PR

**Description:**
Refactors switch `mode` from the checkbox widget into a separate functional widget.
Uses JSX and updates DOM to accomodate Material theme.

Includes Dojo and Material themes.

Resolves #796 

**Dojo Theme**
Jan-03-2020 10-17-15

**Material Theme**
Jan-03-2020 10-18-36

### CODE REVIEW

File: src/switch/index.tsx [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+
+import { formatAriaProperties } from '../common/util';
+import Label from '../label';
+import * as css from '../theme/default/switch.m.css';
+
+interface SwitchProperties {
+	/** Custom aria attributes */
+	aria?: { [key: string]: string | null };
+	/**  Checked/unchecked property of the switch */
+	checked?: boolean;
```
Comment (tomdye) [inline src/switch/index.tsx]: as this is essentially being used as an on/off switch and never a group of checkboxes, how would you feel about changing this to `value`  and removing `checked`

File: src/switch/index.tsx [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+	/** Handler for events triggered by "on over" */
+	onOver?(): void;
+	/** Handler for when the value of the widget changes */
+	onValue?(checked: boolean): void;
+	/** Makes the switch readonly (it may be focused but not changed) */
+	readOnly?: boolean;
+	/** Determines if this input is required, styles accordingly */
+	required?: boolean;
+	/** Determines if this input is valid */
+	valid?: boolean;
```
Comment (tomdye) [inline src/switch/index.tsx]: can this actually be `valid / invalid` ?

File: src/switch/index.tsx [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+	/** The current value */
+	value?: string;
+	/** `id` set on the root button DOM node */
+	widgetId?: string;
```
Comment (tomdye) [inline src/switch/index.tsx]: does this need a `widgetId` property? I do not believe it does

File: src/switch/index.tsx [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+		onValue,
+		onLabel,
+		onOut,
+		onOver,
+		readOnly,
+		required,
+		theme,
+		valid,
+		value,
+		widgetId = uuid()
```
Comment (tomdye) [inline src/switch/index.tsx]: if you remove `widgetId` as a property and remove `uuid`, you can use `id` passed into the factory function as per https://github.com/dojo/widgets/blob/master/src/menu/index.tsx#L289

**Comment by dojo-site** [comment]
This pull request has been deployed to:
* Docs Deployment: https://dojowidgets-ottzckfxd.now.sh

File: src/switch/index.tsx (hunk 3 above)
Comment (ericos) [reply src/switch/index.tsx]: Removed.

**Comment by dojo-site** [comment]
This pull request has been deployed to:
* Docs Deployment: https://dojowidgets-nr0ndpe0v.now.sh
* Tests Deployment: https://dojowidgets-laci5ql39.now.sh

**Comment by tomdye** [comment]
@ericos would like to progress this, should it still be in draft mode?

**Comment by dojo-site** [comment]
This pull request has been deployed to:
* Docs Deployment: https://dojowidgets-qbmiowzwo.now.sh
* Tests Deployment: https://dojowidgets-i8arjj8sf.now.sh

File: src/theme/default/switch.m.css [hunk 5]
```diff
@@ -0,0 +1,49 @@
+@import './variables.css';
+
+.root {
```
Comment (tomdye) [inline src/theme/default/switch.m.css]: can you comment these please?

**Comment by dojo-site** [comment]
This pull request has been deployed to:
* Docs Deployment: https://dojowidgets-60h7k79xl.now.sh
* Tests Deployment: https://dojowidgets-nrnggn3wk.now.sh

**Pull request merged**